### DISCUSSION TITLE
Deploy based on releases (new tags)

### DISCUSSION THREAD

**Discussion by guillempuche** [coollabsio/coolify#4532] [Feature Requests]
Right now Coolify supports deploying of new commits in specific branch.

Guaranteed snapshot. A tag represents an exact snapshot of the codebase at a specific point in time. Deploying from a tag ensures the exact same code is used every time.

No surprises. With branches, there’s always a risk that new commits could modify the branch’s state, leading to inconsistent deployments. Tags provide a clear and fixed reference for logs, audits, or debugging.

**Comment by terion-name** [comment]
so strange that this had no attention =(

**Comment by ChristopheCVB** [comment]
I had the same requirement, ended up implementing it with Github workflows and actions, here is a [Gist](https://gist.github.com/ChristopheCVB/63f05301328dc3a31e0d6ab8cfb65a55) of it.
---
I have a `main` branch that gets auto deployed on push (normal Coolify integration) on a `develop` env and then i have [Release Please](https://github.com/googleapis/release-please) plugged into that branch that creates a release when I merge the Release Please PR created the the [Release Please Github Action](https://github.com/googleapis/release-please-action), and then, a workflow that triggers when a release is created that deploys with a [Github Action](https://github.com/marketplace/actions/coolify-application-deploy) on a Coolify env, `production`, that has auto deploy disabled
